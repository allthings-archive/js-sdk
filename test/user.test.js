import API from '../index'

const client = API(
  process.env.ALLTHINGS_OAUTH_URL,
  process.env.ALLTHINGS_REST_API_URL,
  process.env.ALLTHINGS_OAUTH_CLIENT_ID
)
jest.setTimeout(10000000)

describe('Authentication', () => {
  it('should be able to login', async () => {
    const response = await client({
      method: 'POST',
      path: 'auth/login',
      requiresCsrf: true,
      clientID: true,
      entity: {
        email: process.env.ALLTHINGS_OAUTH_USERNAME,
        password: process.env.ALLTHINGS_OAUTH_PASSWORD,
      },
    })
    expect(typeof response.entity.access_token).toBe('string')
    expect(typeof response.entity.expires_in).toBe('number')
    expect(typeof response.entity.user_id).toBe('string')
  })
  it('should be able to get me', async () => {
    const token = await client({
      method: 'POST',
      path: 'auth/login',
      requiresCsrf: true,
      clientID: true,
      entity: {
        email: process.env.ALLTHINGS_OAUTH_USERNAME,
        password: process.env.ALLTHINGS_OAUTH_PASSWORD,
      },
    })

    const response = await client({
      method: 'GET',
      path: '/api/v1/me',
      accessToken: token.entity.access_token,
    })

    const { lastLogin, ...user } = response.entity

    expect(user).toMatchSnapshot()
    expect(typeof Date.parse(lastLogin)).toBe('number')
  })
})
