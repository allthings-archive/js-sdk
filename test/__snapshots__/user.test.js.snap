// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authentication should be able to get me 1`] = `
Object {
  "_embedded": Object {
    "company": null,
    "notificationSettings": Object {
      "admin-messages": "immediately",
      "deals": "daily",
      "events": "daily",
      "hints-and-tips": "weekly",
      "local-deals": "daily",
      "local-events": "daily",
      "lost-and-found": "daily",
      "messages": "daily",
      "miscellaneous": "weekly",
      "new-things-for-sale": "daily",
      "new-things-to-give": "daily",
      "services": "daily",
      "surveys": "weekly",
    },
    "organization": null,
    "permissions": Array [
      Object {
        "_links": Object {
          "user": Object {
            "href": "/api/v1/users/50f66beaeabc88ab0e000000",
          },
        },
        "id": null,
        "label": null,
        "objectID": "*",
        "objectType": "App",
        "restricted": false,
        "restrictions": Array [],
        "role": "admin",
      },
      Object {
        "_links": Object {
          "user": Object {
            "href": "/api/v1/users/50f66beaeabc88ab0e000000",
          },
        },
        "id": null,
        "label": null,
        "objectID": "*",
        "objectType": "App",
        "restricted": false,
        "restrictions": Array [],
        "role": "article-admin",
      },
      Object {
        "_links": Object {
          "user": Object {
            "href": "/api/v1/users/50f66beaeabc88ab0e000000",
          },
        },
        "id": null,
        "label": null,
        "objectID": "*",
        "objectType": "App",
        "restricted": false,
        "restrictions": Array [],
        "role": "community-article-admin",
      },
    ],
    "profileImage": null,
    "utilisationPeriods": Array [],
  },
  "_links": Object {
    "apps": Object {
      "href": "/api/v1/users/50f66beaeabc88ab0e000000/apps",
    },
    "articles": Object {
      "href": "/api/v1/users/50f66beaeabc88ab0e000000/articles",
    },
    "communityArticles": Object {
      "href": "/api/v1/users/50f66beaeabc88ab0e000000/community-articles",
    },
    "permissions": Object {
      "href": "/api/v1/users/50f66beaeabc88ab0e000000/permissions",
    },
    "self": Object {
      "href": "/api/v1/users/50f66beaeabc88ab0e000000",
    },
    "things": Object {
      "href": "/api/v1/users/50f66beaeabc88ab0e000000/things",
    },
    "utilisationPeriods": Object {
      "href": "/api/v1/users/50f66beaeabc88ab0e000000/utilisation-periods",
    },
  },
  "_meta": Object {
    "company": null,
    "hasPermissions": true,
  },
  "createdAt": "2013-01-16T08:59:22+0000",
  "deleted": false,
  "deletionRequestedAt": null,
  "description": null,
  "email": "admin@allthings.me",
  "emailValidated": true,
  "externalId": null,
  "gender": null,
  "id": "50f66beaeabc88ab0e000000",
  "locale": "en_US",
  "passwordChanged": true,
  "phoneNumber": null,
  "profileImage": null,
  "properties": null,
  "publicProfile": true,
  "receiveAdminNotifications": true,
  "roles": Array [
    "admin",
    "article-admin",
    "community-article-admin",
    "super-admin",
  ],
  "tenantIDs": Object {},
  "type": "allthings_user",
  "username": "admin@allthings.me",
  "vacationSetting": null,
}
`;
